sonar-lits-plugin/src/main/java/com/sonarsource/lits/LitsReport.java

```diff
@@ -0,0 +1,246 @@
+      n += stringSetEntry.getValue().size();
+    }
+
+    this.issuesCount = n;
+  }
+
+  public int issuesCount() {
+    return issuesCount;
+  }
+
+  public List<ResourceReport> resourceReports() {
+    return resourceReports;
+  }
+
+
+  public static class ReportedIssue {
+
+    String message;
+    InputComponent inputComponent;
+    String moduleKey;
+    String ruleKey;
+    String componentKey;
+    String severity;
+    int line;
+
+    public ReportedIssue(String ruleKey, String componentKey, int line, String message, String severity, String moduleKey) {
+      this.ruleKey = ruleKey;
+      this.componentKey = componentKey;
+      this.moduleKey = moduleKey;
+      this.line = line;
+      this.message = message;
+      this.severity = severity;
+    }
+
+    public ReportedIssue(String ruleKey, String componentKey, int line, String message, String severity, InputComponent inputComponent) {
+      this.ruleKey = ruleKey;
+      this.componentKey = componentKey;
+      this.inputComponent = inputComponent;
+      this.line = line;
+      this.message = message;
+      this.severity = severity;
+    }
+
+    public String message() {
+      return message;
+    }
+
+    public String ruleKey() {
+      return ruleKey;
+    }
+
+    public String componentKey() {
+      return componentKey;
+    }
+
+    public int line() {
+      return line;
+    }
+
+    public String severity() {
+      return severity;
+    }
+
+    public int key() {
+      return this.hashCode();
+    }
+  }
+
+
+  public static class ResourceReport {
+
+    Set<ReportedIssue> issues;
+
+    // moduleKey:relative/path/to/file.foo
+    String componentKey;
+
+    InputComponent inputComponent;
+
+    boolean isFile;
+
+    private Map<Integer, List<ReportedIssue>> issuesPerLine = new HashMap<>();
+
+    public ResourceReport(String componentKey, Set<ReportedIssue> issues, FileSystem fileSystem) {
+      this.componentKey = componentKey;
+      this.issues = issues;
+
+      // InputFile corresponding to this resource, it's null for new issues
+      InputComponent inputComponent = null;
+
+      // Module key, first part before colon
+      // It's required to correctly extract relative path from componentKey
+      // It's null for missing issues
+      String moduleKey = null;
+
+      for (ReportedIssue issue : this.issues) {
+        if (issue.inputComponent != null && inputComponent == null) {
+          inputComponent = issue.inputComponent;
+        }
+
+        if (issue.moduleKey != null && moduleKey == null) {
+          moduleKey = issue.moduleKey;
+        }
+
+        if (issuesPerLine.containsKey(issue.line)) {
+          issuesPerLine.get(issue.line).add(issue);
+        } else {
+          List<ReportedIssue> perLine = new ArrayList<>();
+          perLine.add(issue);
+          issuesPerLine.put(issue.line, perLine);
+        }
+      }
+
+      retrieveInputComponent(fileSystem, inputComponent, moduleKey);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "retrieveInputComponent()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)